while-loops/problem-set-while-loops.md

```diff
@@ -0,0 +1,849 @@
+    return next_position
+```
+
+Read through this version and compare to yours. What's the same? What's different? Are there even more ways to approach this problem?
+
+<br />
+
+For extra challenge, consider how we might handle bad inputs, like a missing or empty list. What if the list _does_ contain values that aren't valid indices? Or what if there's a loop in the indices like in `[1, 2, 1]`. Can you write a version of `sidewinder` which instead of returning the end of the chain, returns `True` if it visits _every_ position in the list, and `False` otherwise?
+
+<br />
+
+Be creative!
+
+##### !end-explanation
+
+### !end-challenge
+
+<!-- ======================= END CHALLENGE ======================= -->
+<!-- prettier-ignore-end -->
+
+### A Little Helper for the Next Question
+
+The following coding question uses a helper function called `input_int`. It works just like the [`input`](https://docs.python.org/3/library/functions.html#input) function that we've seen before, except that instead of returning a string, we will assume that it returns an integer.
+
+In the question block below, we don't need to define `input_int`. We can assume that it is available, the same way `input` is available.
+
+To experiment with the coding question in VS Code or repl.it, we will need an implementation for `input_int`. In Python, safely converting from a string to an integer requires handling exceptions. We haven't discussed that material yet, so here is a working implementation.
+
+<details style="max-width: 700px; margin: auto;">
+    <summary>A working implementation of `input_int`</summary>
+```python
+# accept a variable number of positional parameters
+def input_int(*args):
+    # loop forever (until we get valid input)
+    while True:
+        # get string input
+        user_input = input(*args)
+        try:
+            # try converting to an int (will return if successful)
+            return int(user_input)
+
+            # if the conversion succeeded, we will have returned already
+        except ValueError:
+            # could not convert the input, so show an error message
+            print('Try again.')
+```
+</details>
+
+
+<!-- prettier-ignore-start -->
+<!-- >>>>>>>>>>>>>>>>>>>>>> BEGIN CHALLENGE >>>>>>>>>>>>>>>>>>>>>> -->
+### !challenge
+
+* type: code-snippet
+* language: python3.6
+* id: 0fdaf2f7-0573-4e0a-8fd2-46f15076de24
+* title: While Loops
+
+##### !question
+
+Let's sum some numbers! We will write a function called `silly_sum` that reads a series of numbers from user input and returns the sum. To keep things interesting, let's also decide that entering `0` means the user is done entering input, and that we'll also stop asking for further input if the sum reaches or exceeds `1000`.
+
+Provide an implementation for `silly_sum` so that it
+- reads numbers from the user (use `input_int`) summing as we go until either
+    - the user enters `0`, or
+    - the sum reaches or exceeds `1000`
+
```

Comment:
I've personally enjoyed and treasured providing the tests in `pytest` form whenever there's a code challenge. Is it possible for you to do that too?

Reply:
From my response above, I'm not really inclined to add the tests, it _really_ doesn't help for silly_sum, and it felt weird to do it for one but not the other.